### PR TITLE
Refactor ensureConfigurationConfigMapsExist to reduce duplicate code

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

ensureConfigurationConfigMapsExist is used in pipelinerun, taskrun, run test files to ensure the test data contains needed configmap and they all share the same code. This commit move this function to a test package and can be used accross different files.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by Yongxuanzhang** [comment]
The function cannot be moved to pkg/reconciler/testing due to import cycle.

File: test/controller.go [hunk 1]
```diff
@@ -323,3 +325,55 @@ var _ ktesting.Reactor = (*ResourceVersionReactor)(nil)
 func PrependResourceVersionReactor(f *ktesting.Fake) {
 	f.ReactionChain = append([]ktesting.Reactor{&ResourceVersionReactor{}}, f.ReactionChain...)
 }
+
+// EnsureConfigurationConfigMapsExist makes sure all the configmaps exists.
+func EnsureConfigurationConfigMapsExist(d *Data) {
```
Comment (JeromeJu) [inline test/controller.go:330]: Thanks Yongxuan, shall we also refactor for the following file https://github.com/tektoncd/pipeline/blob/99226de6ef1c3f14e38e9b8cbc1fe0e60753a079/pkg/resolution/resolver/framework/testing/fakecontroller.go/#L123?

Comment (abayer) [reply test/controller.go:330]: No - that's specifically for the resolver config maps in the `tekton-pipelines-resolvers` namespace.

**Pull request merged**